src/components/form/form.js

```diff
@@ -332,12 +359,20 @@ class Form extends React.Component {
     if (!nextProps.stickyFooter && this.props.stickyFooter) {
       this.removeStickyFooterListeners();
     }
+
+    if (nextProps.unsavedWarning) {
```

Comment:
you should check here if `this.props.unsavedWarning` is different, otherwise it will add listener after listener after listener